src/realm/object-store/results.hpp

```diff
@@ -156,6 +156,11 @@ class Results {
     // Create a new Results by removing duplicates based on the specified key paths.
     Results distinct(std::vector<std::string> const& keypaths) const REQUIRES(!m_mutex);
 
+    // Create a new Results by filtering using a user supplied function.
+    // The user supplied function can be called from any thread, so it has
+    // to be a pure function or at least thread safe.
+    Results filter_by_method(FilterDescriptor&& predicate) const REQUIRES(!m_mutex);
```

Comment:
It would be nice with a convenience method that takes the function pointer directly as well (similar to most of the descriptor methods on Results).

Reply:
I guess there is no need to have both. Now you got your function taking the function as parameter directly.